frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -695,3 +695,11 @@ export type QueryBrowserProps = {
   timespan?: number;
   units?: string;
 };
+
+export type StorageClass = K8sResourceCommon & {
+  provisioner: string;
+  parameters: object;
+  reclaimPolicy?: string;
+  volumeBindingMode?: string;
+  allowVolumeExpansion?: boolean;
```

Comment:
Are these top-level properties on the resource or should they be under `spec`?

Reply:
These are top level properties for SC. 
```
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  creationTimestamp: "2023-06-28T04:32:04Z"
  name: gp2-csi
  resourceVersion: "5232"
  uid: dbb00cda-99d0-4638-8b05-88602ccdab58
parameters:
  encrypted: "true"
  type: gp2
provisioner: ebs.csi.aws.com
reclaimPolicy: Delete
volumeBindingMode: WaitForFirstConsumer
```